src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs

```diff
@@ -553,8 +553,14 @@ public override Task<CompletionDescription> GetDescriptionAsync(Document documen
                 : Task.FromResult(CompletionDescription.Empty);
         }
 
-        public override bool ShouldTriggerCompletion(
-            SourceText text, int caretPosition, CompletionTrigger trigger, ImmutableHashSet<string> roles = null, OptionSet options = null)
+        public override bool ShouldTriggerCompletion(SourceText text, int caretPosition, CompletionTrigger trigger, ImmutableHashSet<string> roles = null, OptionSet options = null)
+        {
+            var document = text.GetOpenDocumentInCurrentContextWithChanges();
+            return ShouldTriggerCompletion(document?.Project, text, caretPosition, trigger, roles, options);
```

Comment:
you potentialy pass in null to ShouldTriggerCmopletion, however, below i see that code do project.LanguageServices.  Won't this crash?